src/Schema/Types/LaravelEnumType.php

```diff
@@ -41,8 +41,8 @@ public function __construct(string $enumClass, ?string $name = null)
                 function (Enum $enum): array {
                     return [
                         'name' => $enum->key,
-                        'value' => $enum,
-                        'description' => "$enum->value",
+                        'value' => $enum->value,
```

Comment:
The internal value should stay an instance of the enum class. That makes it work smoothly with type hints and model casts.

Reply:
But it would record instance as JSON when column type is JSON or TEXT

Reply:
Not if attribute casting is used (https://github.com/BenSampo/laravel-enum#attribute-casting).

Anyways, that has been the current behaviour for quite a while. We can't break it in a minor release. You can always use your own version of `LaravelEnumType` if you need to change that in your project.